Test/Unit/Block/Price/DetailsTest.php

```diff
@@ -0,0 +1,251 @@
+        $this->sut->setSaleableItem($saleableItemMock);
+        $this->assertSame(null, $this->sut->getData('tax_rate'));
+    }
+
+    public function testGetFormattedTaxRate(): void
+    {
+        $this->sut->setData('tax_rate', '19');
+        $this->assertInstanceOf(Phrase::class, $this->sut->getFormattedTaxRate());
+    }
+
+    public function testGetFormattedTaxRateIsZero(): void
+    {
+        $saleableItemMock = $this->getMockBuilder(SaleableInterface::class)
+            ->setMethods(['getTaxPercent'])
+            ->disableOriginalConstructor()
+            ->getMockForAbstractClass();
+        $saleableItemMock
+            ->expects($this->at(0))
+            ->method('getTaxPercent')
+            ->willReturn(7);
+
+        $this->sut->setSaleableItem($saleableItemMock);
+        $this->assertInstanceOf(Phrase::class, $this->sut->getFormattedTaxRate());
+    }
+
+    public function testGetFormattedTaxRateIsFive(): void
+    {
+        $saleableItem2Mock = $this->getMockBuilder(SaleableInterface::class)
+            ->setMethods(['getTaxPercent', 'getTaxClassId'])
+            ->disableOriginalConstructor()
+            ->getMockForAbstractClass();
+        $saleableItem2Mock
+            ->expects($this->at(0))
+            ->method('getTaxPercent')
+            ->willReturn(null);
+
+        $saleableItem2Mock
+            ->expects($this->at(1))
+            ->method('getTaxClassId')
+            ->willReturn('simple');
+
+        $storeMock = $this->getMockBuilder(\Magento\Store\Api\Data\StoreInterface::class)
+            ->disableOriginalConstructor()
+            ->getMockForAbstractClass();
+
+        $this->storeManagerMock
+            ->expects($this->at(0))
+            ->method('getStore')
+            ->willReturn($storeMock);
+
+        $this->customerSessionMock
+            ->expects($this->at(0))
+            ->method('getCustomerGroupId')
+            ->willReturn(10);
+        $groupMock = $this->createMock(\Magento\Customer\Model\Data\Group::class);
+
+        $groupMock
+            ->expects($this->at(0))
+            ->method('getTaxClassId')
+            ->willReturn(20);
+
+        $this->groupRepositoryMock
+            ->expects($this->at(0))
+            ->method('getById')
+            ->withAnyParameters(10)
+            ->willReturn($groupMock);
+
+        $dataMock = $this->createMock(\Magento\Framework\DataObject::class);
+
+        $dataMock
+            ->expects($this->at(0))
+            ->method('setData')
+            ->willReturn($dataMock);
+
+        $this->taxCalculationMock
+            ->expects($this->at(0))
+            ->method('getRateRequest')
+            ->willReturn($dataMock);
+
+        $this->taxCalculationMock
+            ->expects($this->at(1))
+            ->method('getRate')
+            ->willReturn(5);
+
+        $this->sut->setSaleableItem($saleableItem2Mock);
+        $this->assertInstanceOf(Phrase::class, $this->sut->getFormattedTaxRate());
```

Comment:
Should we add a more concrete assertion (applies to all `$this->assertInstanceOf(Phrase::class, ...)` cases)?

Reply:
Yes you are right. I have changed it.